### PR TITLE
add check_usb_hub_connected to utils

### PR DESCRIPTION
**Pull request by hhatpsi**

### CODE REVIEW

File: src/pnpq/devices/utils.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+from serial.tools.list_ports import comports as list_comports
+import logging
+
+logger = logging.getLogger("utils")
+
+USB_HUB_iVENDOR = "2109"
```
Comment (zigen) [inline src/pnpq/devices/utils.py:6]: `USB_HUB_VENDOR_ID` is fine. I don't want to put `i` in the constant name.

File: src/pnpq/devices/utils.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+USB_HUB_iPROFUCT = ["0817", "2817"]
```
Comment (zigen) [inline src/pnpq/devices/utils.py:7]: the number of available usb hubs will grow. so, we should check by the tuple of vid and pid.
then the available usb hubs are defined like this
```py
AVAILABLE_USB_HUBS: list[tuple[str, str]] = [
  ("2109", "0817") # some usb hub name
  ("2109", "2817") # another usb hub name
]
```

File: src/pnpq/devices/waveplate_thorlabs_kb10crm.py [hunk 3]
```diff
@@ -61,7 +64,9 @@ def __init__(
         self.home_timeout = 20
         self.max_channel = 1
         self.auto_update = False
+        self.hub = check_usb_hub_connected()
```
Comment (zigen) [inline src/pnpq/devices/waveplate_thorlabs_kb10crm.py:67]: should be `self.hub_connected`. `self.hub` looks this represents a usb hub instance.

File: src/pnpq/devices/utils.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    logger.debug(f"get_available_port(serial_number: {device_serial_number})")
+    available_ports = list_comports()
+    for port in available_ports:
+        if port.serial_number == device_serial_number:
+            logger.debug(f"port found: {port.device}")
+            return port.device
+    return None
+
+
+def check_usb_hub_connected() -> bool | None:
```
Comment (zigen) [inline src/pnpq/devices/utils.py:20]: this returns only bool value. this must be `-> bool`

File: src/pnpq/utils.py [hunk 5]
```diff
@@ -8,7 +9,23 @@ def get_available_port(device_serial_number: str) -> str | None:
...
     available_ports = list_comports()
     for port in available_ports:
+        logger.debug(f"port: {port.device}")
         if port.serial_number == device_serial_number:
             logger.debug(f"port found: {port.device}")
             return port.device
     return None
+
+
+def check_usb_hub_connected() -> bool | None:
```
Comment (zigen) [inline src/pnpq/utils.py:19]: this also always return boolean value.

**Review by zigen — CHANGES_REQUESTED** [review]
please add pyproject.toml and requirements*.lock

File: src/pnpq/devices/utils.py (hunk 1 above)
Comment (hhatpsi) [reply src/pnpq/devices/utils.py]: AVAILABLE_USB_HUBS was defined to be a list of (vendor, product) pairs.

File: src/pnpq/utils.py (hunk 5 above)
Comment (hhatpsi) [reply src/pnpq/utils.py]: method annotation was modified.

**Review by hhatpsi — COMMENTED** [review]
remove os-dependent code for detecting usb hub and redefine check_usb_hub_connected method based on usb library.

**Review by zigen — APPROVED** [review]

**Pull request merged**